### PR TITLE
Fix - 502 in save layout is abruptly taking me to an error screen

### PR DESCRIPTION
**Pull request by jsartisan**
## Description
Server error on any API on editor page was taking user to error page which is not ideal. So removed the crashing on API failures. Now will be crashing the page manually on certain errors.

Fixes # (issue)
> #2452 

## Type of change
- Bug fix

## How Has This Been Tested?

- Manually tested

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by hetunandu** [comment]
We also want to show the error page when the /me call fails since this is always the first call.

File: app/client/src/api/Api.tsx [hunk 1]
```diff
@@ -71,15 +71,16 @@ axiosInstance.interceptors.response.use(
...
-      });
-    }
+
     if (error.response) {
+      if (error.response.status === API_STATUS_CODES.SERVER_ERROR) {
+        return Promise.reject({
+          ...error,
+          code: ERROR_CODES.SERVER_ERROR,
+          message: ERROR_500,
+        });
```
Comment (hetunandu) [inline app/client/src/api/Api.tsx:81]: Can we add a crash for 502 in `/me` call

**Review by hetunandu — CHANGES_REQUESTED** [review]

**Comment by hetunandu** [comment]
@all-contributors add @jsartisan  for code

File: app/client/src/sagas/ErrorSagas.tsx [hunk 2]
```diff
@@ -207,7 +208,10 @@ export default function* errorSagas() {
     flushErrorsAndRedirectSaga,
   );
   yield takeLatest(
-    ReduxActionTypes.SAFE_CRASH_APPSMITH_REQUEST,
+    [
+      ReduxActionErrorTypes.FETCH_USER_DETAILS_ERROR,
```
Comment (jsartisan) [inline app/client/src/sagas/ErrorSagas.tsx:212]: @hetunandu This will set the safeCrash now for user init call.

Comment (hetunandu) [reply app/client/src/sagas/ErrorSagas.tsx:212]: Not a clean solve. Why cant you dispatch safe crash from the user saga itself?

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/sagas/userSagas.tsx [hunk 3]
```diff
@@ -118,6 +119,13 @@ export function* getCurrentUserSaga() {
         error,
       },
     });
+
+    yield put({
```
Comment (jsartisan) [inline app/client/src/sagas/userSagas.tsx:123]: @hetunandu Dispatched the safeCrash action from here.

**Review by hetunandu — APPROVED** [review]

**Pull request merged**